pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -301,3 +304,27 @@ type PipelineRunList struct {
 type PipelineTaskRun struct {
 	Name string `json:"name,omitempty"`
 }
+
+// PipelineRunTaskSpec  can be used to configure specific
+// specs for a concrete Task
+type PipelineRunTaskSpec struct {
+	TaskName               string       `json:"taskName,omitempty"`
+	TaskServiceAccountName string       `json:"serviceAccountName,omitempty"`
+	TaskPodTemplate        *PodTemplate `json:"podTemplate,omitempty"`
+}
+
+// GetTaskSpec returns the task specific spec for a given
+// PipelineTask if configured, otherwise it returns the PipelineRun's default.
+func (pr *PipelineRun) GetTaskSpec(pipelineTaskName string) (string, *PodTemplate) {
```

Comment:
I'm not too sure if this is exactly possible but can we share the code between the v1alpha1 and v1beta1 types instead of copying them?

Reply:
We might at some point but this is orthogonal to this PR. I am gonna look at how the conversion webhook would help us on this.

Reply:
Ok, then i keep it in both so far.